tests/FSharp.Compiler.ComponentTests/CompilerCompatibilityTests.fs

```diff
@@ -0,0 +1,97 @@
+        let globalJsonContent = $"""{{
+  "sdk": {{
+    "version": "{targetFramework}"
+  }}
+}}"""
+        let globalJsonPath = Path.Combine(projectPath, "global.json")
+        File.WriteAllText(globalJsonPath, globalJsonContent)
+        globalJsonPath
+    
+    let cleanupGlobalJson globalJsonPath =
+        if File.Exists(globalJsonPath) then
+            File.Delete(globalJsonPath)
+    
+    let runDotnetBuild projectPath compilerVersion =
+        let globalJsonPath = 
+            match compilerVersion with
+            | "local" -> 
+                // For local compiler, use LoadLocalFSharpBuild property
+                None
+            | version -> 
+                // For specific .NET versions, create global.json
+                Some (createGlobalJson version projectPath)
+        
+        let args = 
+            match compilerVersion with
+            | "local" -> "build -c Release -p:LoadLocalFSharpBuild=True"
+            | _ -> "build -c Release"
+                
+        let (exitCode, output, error) = Commands.executeProcess "dotnet" args projectPath
+        
+        // Cleanup global.json if created
+        globalJsonPath |> Option.iter cleanupGlobalJson
+        
+        if exitCode <> 0 then
+            let outputStr = String.concat "\n" output
+            let errorStr = String.concat "\n" error
+            failwith $"Build failed with exit code {exitCode}. Output: {outputStr}. Error: {errorStr}"
+        
+        String.concat "\n" output
+    
+    let runApp appBinaryPath =
+        let (exitCode, output, error) = Commands.executeProcess "dotnet" appBinaryPath (Path.GetDirectoryName(appBinaryPath))
+        (exitCode, String.concat "\n" output, String.concat "\n" error)
+    
+    let cleanBinObjDirectories projectPath =
+        let binPath = Path.Combine(projectPath, "bin")
+        let objPath = Path.Combine(projectPath, "obj")
+        
+        if Directory.Exists(binPath) then
+            Directory.Delete(binPath, true)
+        if Directory.Exists(objPath) then
+            Directory.Delete(objPath, true)
+
+    let getAppDllPath () =
+        // The app is built to artifacts directory due to Directory.Build.props
+        Path.Combine(__SOURCE_DIRECTORY__, "..", "..", "artifacts", "bin", "CompilerCompatApp", "Release", "net8.0", "CompilerCompatApp.dll")
+
+    [<Theory>]
+    [<InlineData("local", "local", "Baseline scenario - Both library and app built with local compiler")>]
+    [<InlineData("10.0.100-rc.1.25411.109", "local", "Forward compatibility - Library built with .NET 10 SDK, app with local compiler")>]
+    [<InlineData("local", "10.0.100-rc.1.25411.109", "Backward compatibility - Library built with local compiler, app with .NET 10 SDK")>]
+    member _.``Compiler compatibility test``(libCompilerVersion: string, appCompilerVersion: string, scenarioDescription: string) =
+        // Clean previous builds
+        cleanBinObjDirectories libProjectPath
```

Comment:
The hardcoded long version will not work.
Make use of:
"local"
"latest" (just leave it)
"9" (will use 9)

Then cover **all** combinations which have "local" at least on one of the sides.